utilities/transactions/optimistic_transaction.cc

```diff
@@ -54,6 +56,22 @@ Status OptimisticTransaction::Prepare() {
 }
 
 Status OptimisticTransaction::Commit() {
+  auto txn_db_impl = static_cast_with_check<OptimisticTransactionDBImpl,
+                                            OptimisticTransactionDB>(txn_db_);
+  assert(txn_db_impl);
+  OccValidationPolicy policy = txn_db_impl->GetValidatePolicy();
```

Comment:
Not big deal, but I feel it slightly more readable by dong a `switch(txn_db_impl->GetValidatePolicy()) { case...`.
Considering the time zone difference, I'm not going to hold the PR because of it. But if you are interested, I encourage you to send a follow up PR for it.

Reply:
ummm, switch/case here seems more elegant.
I will make another pr in this week.